File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/GetFavoriteRecipesUseCaseTest.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.usecase.GetFavoriteRecipesUseCase;
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Java6Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
public class GetFavoriteRecipesUseCaseTest {
    @Mock private RecipeRepository recipeRepository;
    private GetFavoriteRecipesUseCase subject;

    @Before
    public void setUp() {
        subject = new GetFavoriteRecipesUseCase(recipeRepository);
    }

    @Test
    public void execute() {
        List<Recipe> recipes = Arrays.asList(
                Recipe.builder().build(),
                Recipe.builder().build(),
                Recipe.builder().build(),
                Recipe.builder().build()
        );

        given(recipeRepository.findAllFavorite()).willReturn(recipes);

        assertThat(subject.execute()).isEqualTo(recipes);
    }
}

File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/GetFavoriteRecipesUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*

class GetFavoriteRecipesUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()
    private lateinit var subject: GetFavoriteRecipesUseCase

    @BeforeEach
    fun setUp() {
        subject = GetFavoriteRecipesUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        val recipes = Arrays.asList(
            Recipe(),
            Recipe(),
            Recipe(),
            Recipe()
        )

        whenever(recipeRepository.findAllFavorite()).thenReturn(recipes)

        assertEquals(recipes, subject.execute())
    }
}